scipy/optimize/tests/test__gensa.py

```diff
@@ -0,0 +1,90 @@
+"""
+Unit tests for the Generalized Simulated Annealing global optimizer
+"""
+from scipy.optimize import _gensa
+from scipy.optimize._gensa import GenSARunner
+import numpy as np
+from numpy.testing import (assert_equal, TestCase, assert_allclose,
+                           assert_almost_equal, assert_raises)
+
+
+class TestGenSA(TestCase):
+
+    def setUp(self):
+        # Using Rastrigin function for performing tests
+        self.func = lambda x: np.sum(x * x - 10 * np.cos(2 * np.pi * x))\
+                + 10 * np.size(x)
+        self.weirdfunc = lambda x: 1e15
+        self.ld_bounds = [(-5.12, 5.12)] * 2
+        self.hd_bounds = self.ld_bounds * 5
+        self.nbtestvalues = 5000
+        self.defautgr = (self.func, None, self.ld_bounds)
+
+    def tearDown(self):
+        pass
+
+    def test_low_dim(self):
+        ret = _gensa.gensa(self.func, None, self.ld_bounds)
+        assert_allclose(ret.fun, 0., atol=1e-12)
+
+    def test__visiting_dist(self):
+        gr = GenSARunner(*(self.defautgr))
+        values = np.zeros(self.nbtestvalues)
+        for i in np.arange(self.nbtestvalues):
+            values[i] = gr._visita()
+        # Visiting distribution is a distorted version of Cauchy-Lorentz
+        # distribution, and as no 1st and higher moments (no mean defined,
+        # no variance defined).
+        # Check that big tails values are generated
+        assert(np.min(values) < 1e-10)
```

Comment:
plain ``assert`` can be optimized away during byte-compiling, so we don't use it in scipy. please use ``assert_`` from ``np.testing`` instead